crml/cennzx/src/lib.rs

```diff
@@ -223,23 +238,23 @@ decl_module! {
 				Error::<T>::CannotAddLiquidityWithZero
 			);
 			ensure!(
-				T::AssetSystem::free_balance(core_asset_id, &from_account) >= core_amount,
+				T::MultiCurrency::free_balance(&from_account, Some(core_asset_id.clone())) >= core_amount,
```

Comment:
nit: is asset_id copy? might not need all these clones

Reply:
Yes, it is. I will get rid of the clones.

Reply:
Done